### ISSUE TITLE
[a11y]: API Key Modal, Unable to close dialog using esc

### ISSUE THREAD

**Issue by anamikaanu96** [carbon-design-system/ibm-products#6252]
### Package

@carbon/ibm-products

### Browser

Firefox

### Operating System

Windows

### Package version

v2.52.0-rc.0

### Automated testing tool and ruleset

Manual testing

### Assistive technology

JAWS

### Description

After editing & saving the key, "API Key Successfully saved" heading replaces "Save an API key" haeding. Unable to close this second dialog using esc

Expected behaviour : The Modal expected to close with a press of Escape key 

### Reproduction/example

https://ibm-products.carbondesignsystem.com/?path=/story/ibm-products-components-generating-an-api-key-apikeymodal--generate&globals=viewport:basic

### Steps to reproduce

This was reported in API Key Modal - Edit, Edit with Error

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems
